compiler/ballerina-parser/src/main/java/io/ballerina/compiler/internal/parser/BallerinaParser.java

```diff
@@ -12860,7 +12860,14 @@ private STNode parseTypeDescOrExpr() {
                     return parseActionOrExpressionInLhs(STNodeFactory.createEmptyNodeList());
                 }
 
-                return parseTypeDescriptor(ParserRuleContext.TYPE_DESC_IN_TYPE_BINDING_PATTERN);
+                STNode typedesc =  parseTypeDescriptor(ParserRuleContext.TYPE_DESC_IN_TYPE_BINDING_PATTERN);
+                // Tuple type desc can contain rest descriptor which is not a regular type desc,
+                // hence handle it separately.
+                if (peek().kind == SyntaxKind.ELLIPSIS_TOKEN) {
+                    STNode ellipsis = parseEllipsis();
```

Comment:
since we check the kind can consume directly. 
Also the method we are in is parseTypeDescOrExpr. Is it only called within a tuple?
Otherwise will have to do an additional check?

Reply:
Updated, thanx